src/Microsoft.Azure.SignalR.Common/Utilities/Rest/BinaryPayloadMessageContent.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.IO;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading.Tasks;
+using MessagePack;
+using Microsoft.AspNetCore.SignalR.Protocol;
+
+namespace Microsoft.Azure.SignalR.Common
+{
+    internal class BinaryPayloadMessageContent : HttpContent
+    {
+        private static readonly Dictionary<string, byte[]> ProtocolMap = new Dictionary<string, byte[]>(2)
+        {
+            {Constants.Protocol.Json, Encoding.UTF8.GetBytes(Constants.Protocol.Json) },
+            {Constants.Protocol.MessagePack,Encoding.UTF8.GetBytes(Constants.Protocol.MessagePack)}
+        };
+        private static readonly MediaTypeHeaderValue ContentType = new("application/octet-stream");
+
+        private readonly PayloadMessage _payloadMessage;
+        private readonly IReadOnlyList<IHubProtocol> _hubProtocols;
+
+        public BinaryPayloadMessageContent(PayloadMessage payloadMessage, IReadOnlyList<IHubProtocol> hubProtocols)
+        {
+            _payloadMessage = payloadMessage ?? throw new ArgumentNullException(nameof(payloadMessage));
+            _hubProtocols = hubProtocols ?? throw new ArgumentNullException(nameof(hubProtocols));
+            Headers.ContentType = ContentType;
+        }
+
+        protected override async Task SerializeToStreamAsync(Stream stream, TransportContext context)
+        {
+            var memoryBufferWriter = MemoryBufferWriter.Get();
+            try
+            {
+                WriteMessageCore(memoryBufferWriter);
+                await memoryBufferWriter.CopyToAsync(stream);
```

Comment:
MemoryBufferWriter._cacheInstance is thread local, so using await inside makes `Return` to another thread

Reply:
As memory buffer writer actually doesn't hold any byte array after `Reset()`, I think it's safe to directly new a buffer writer. Will update.

Reply:
updated